frontend/src/main.jsx

```diff
@@ -1,13 +1,28 @@
-import React from "react";
+import { Suspense, StrictMode, lazy } from "react";
 import ReactDOM from "react-dom/client";
-import App from "./App.jsx";
 import "./index.css";
 import { BrowserRouter } from "react-router-dom";
+import { QueryClientProvider, QueryClient } from "@tanstack/react-query";
+import { AppLoader } from "./suspense/AppLoader";
+import ErrorBoundary from "./suspense/ErrorBoundary";
+import { ErrorPage } from "./suspense/ErrorPage";
 
-ReactDOM.createRoot(document.getElementById("root")).render(
-  <React.StrictMode>
-    <BrowserRouter>
-      <App />
-    </BrowserRouter>
-  </React.StrictMode>,
+const App = lazy(() => import("./App"));
+
+const root = ReactDOM.createRoot(document.getElementById("root"));
+
+const queryClient = new QueryClient();
+
+root.render(
+  <StrictMode>
+    <ErrorBoundary fallback={<ErrorPage />}>
```

Comment:
if there's a server error wouldn't this hide the entire app? can you think of a way to structure this so it only hides portions that are failing?

Reply:
I'll add more component-local error handling in the frontend